backend/typescript/middlewares/validators/petValidators.ts

```diff
@@ -0,0 +1,168 @@
+import { Request, Response, NextFunction } from "express";
+import { getApiValidationError, validatePrimitive, validateEnum } from "./util";
+import { PetStatus, Sex } from "../../types";
+
+const petStatusEnum: PetStatus[] = [
```

Comment:
Is there a way to make this generated from the PetStatus in types.ts? So that there's a single point of truth for the petStatus.

If it's a lot of work then it's not a big deal

Reply:
https://stackoverflow.com/questions/66751035/generate-a-string-array-from-a-union-type-typescript

I don't think there's a way to generate an array using a union type, so instead I just defined the values and generated the type and array from that. Let me know if it's too cluttered now or anything

Reply:
Hmm yeah that's a good work around, now we have all the statuses in one place. 

Looks good! mergeeee